include/zephyr/arch/common/arch_inlines.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Meta Platforms.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_ARCH_COMMON_ARCH_INLINES_H_
+#define ZEPHYR_INCLUDE_ZEPHYR_ARCH_COMMON_ARCH_INLINES_H_
+
+#ifndef ZEPHYR_INCLUDE_ARCH_INLINES_H_
+#error "This header shouldn't be included directly"
+#endif /* ZEPHYR_INCLUDE_ARCH_INLINES_H_ */
+
+#ifndef _ASMLANGUAGE
+
+#include <zephyr/kernel_structs.h>
+
+#ifndef CONFIG_ARCH_HAS_CUSTOM_CURRENT_IMPL
+static ALWAYS_INLINE struct k_thread *arch_curr_thread(void)
```

Comment:
why not spell it out, are we trying to save on charachters here? `arch_current_thread()`

Reply:
So it looks more consistent with `arch_curr_cpu()` in the [`arch_inlines.h`](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/arch/riscv/arch_inlines.h)

Reply:
that is something we can always change to also use `current`.

Reply:
The latest version uses `arch_current_thread()`